File: bms/go/geo/geo_r3_PreciseVectorFromVector/geo_r3_PreciseVectorFromVector.go
package test

import "math/big"

var ()

const (
	geo_r3_prec = big.MaxPrec
)

func geo_r3_PreciseVectorFromVector(v geo_r3_Vector) geo_r3_PreciseVector {
	return geo_r3_NewPreciseVector(v.X, v.Y, v.Z)
}

func geo_r3_NewPreciseVector(x, y, z float64) geo_r3_PreciseVector {
	return geo_r3_PreciseVector{
		X: geo_r3_precFloat(x),
		Y: geo_r3_precFloat(y),
		Z: geo_r3_precFloat(z),
	}
}

func geo_r3_precFloat(f float64) *big.Float {
	return new(big.Float).SetPrec(geo_r3_prec).SetFloat64(f)
}

type geo_r3_Vector struct {
	X, Y, Z float64
}

type geo_r3_PreciseVector struct {
	X, Y, Z *big.Float
}

File: translations_only/fail/go/geo/geo_r3_PreciseVectorFromVector/gemini/Hinted/geo_r3_PreciseVectorFromVector.rs

use std::f64::consts::PI;

#[derive(Debug)]
struct GeoR3Vector {
    x: f64,
    y: f64,
    z: f64,
}

#[derive(Debug)]
struct GeoR3PreciseVector {
    x: f64,
    y: f64,
    z: f64,
}

fn geo_r3_precise_vector_from_vector(v: GeoR3Vector) -> GeoR3PreciseVector {
    GeoR3PreciseVector {
        x: v.x,
        y: v.y,
        z: v.z,
    }
}

fn geo_r3_new_precise_vector(x: f64, y: f64, z: f64) -> GeoR3PreciseVector {
    GeoR3PreciseVector { x, y, z }
}

fn geo_r3_prec_float(f: f64) -> f64 {
    f
}

fn main() {
    let v = GeoR3Vector {
        x: 1.00795190635e-313,
        y: 9.36221502393658e-97,
        z: 6.679571501111919e-308,
    };
    let pv = geo_r3_precise_vector_from_vector(v);
    println!("{:?}", pv);
}
